Marlin/src/module/ft_motion/resonance_generator.h

```diff
@@ -0,0 +1,73 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (c) 2025 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (c) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
+
+#include "../../inc/MarlinConfigPre.h"
+
+#include <math.h>
+
+typedef struct FTMResonanceTestParams {
+  AxisEnum axis       = NO_AXIS_ENUM; // Axis to test
+  float min_freq      = 5.0f;         // Minimum frequency [Hz]
+  float max_freq      = 100.0f;       // Maximum frequency [Hz]
+  float octave_duration = 40.0f;      // Octave duration for logarithmic progression
+  float accel_per_hz  = 60.0f;        // Acceleration per Hz [mm/sec/Hz] or [g/Hz]
```

Comment:
Here's where i don't understand why not use constant amplitude in mm. In my tests it works better. Or is this something specific to accelerometer measurement?

Reply:
With constant amplitude I don't meant that it is not configurable, I mean that it is not frequency dependant:
Traj=amplitude*sine(t*2pi*freq)

Instead of accel_per_hz and correction factor, just a single "amplitude "